validate-data.js

```diff
@@ -0,0 +1,212 @@
+    return null;
+}
+
+/**
+ * Whether obj has been released or not
+ * If releaseDate has a days part i.e. is `YYYY-MM-DD`, then its assumed it has been released
+ */
+function hasBeenReleased(obj) {
+    return dateStrHasDayPart(obj.releaseDate);
+}
+
+/**
+ * Print an error to console and exit with a status code of 1
+ */
+function error(message) {
+    console.error(`${scope} ${version} ${message}`);
+    process.exit(1);
+}
+
+// Globally scoped variables used in error()
+let scope = '';
+let version = '';
+
+// Validate data
+for (var i = 0; i < data.data.length; i++) {
+    // The current obj being processed e.g. the obj with version 5.4
+    const currObj = data.data[i];
+    // The prev obj, e.g. 5.3
+    const prevObj = i > 0 ? data.data[i - 1] : null;
+    // The next obj, e.g. 6.0
+    const nextObj = i < (data.data.length - 1) ? data.data[i + 1] : null;
+    // The initial version 2 majors ahead e.g. for 4.13 this will be 6.0, for 5.2 this will be 7.0
+    const dmajObj = getVersionObj((getMajor(currObj) + 2) + '.0');
+    const isFinalMinor = nextObj && getMajor(currObj) !== getMajor(nextObj);
+    // Note that this is first minor for a major in data.json, not necessarily the actual first version
+    // For instance 5.2 might be the first minor in data.json, though 5.0 was the actual first version
+    // The !isFinalMinor check is for something like 4.13 which is the only version for CMS 4
+    const isFirstMinor = !isFinalMinor && (!prevObj || getMajor(prevObj) !== getMajor(currObj));
+    const isIntermediateMinor = !isFirstMinor && !isFinalMinor;
+
+    // variable which is reused for mutliple assertions
+    let expected;
+
+    // Update globally scope objected
+    version = currObj.version;
+
+    // Validate releaseDate - applies to all scenarios
+    if (hasBeenReleased(currObj)) {
+        scope = 'Released version';
+        if (currObj.releaseDateExtra !== null) {
+            error(`releaseDateExtra must be null`);
+        }
+    } else {
+        scope = 'Unlreleased version';
+        if (!currObj.releaseDate.match(/^(\d{4}-\d{2})$/)) {
+            error(`releaseDate must be in YYYY-MM format`);
+        }
+    }
+
+    // The following has some duplicated logic and extra nesting due to prioritising
+    // readabilitiy over conciseness
+    if (isFirstMinor) {
+        // e.g. 6.0, 7.0
+        scope = 'First minor version';
+        if (currObj.supportLength !== 'Standard') {
+            error(`supportLength must be "Standard"`);
+        }
+        // Standard + Full support - Approximately 6 months, until the next minor release is performed
+        if (nextObj) {
+            // Here we use "until the next minor release is performed"
+            if (currObj.partialSupport !== nextObj.releaseDate) {
+                error(`partialSupport must be "${nextObj.releaseDate}"`);
+            }
+        } else {
+            // Here we use "approximately 6 months", because there is no known next minor release date
```

Comment:
```suggestion
            // Here we use "approximately 6 months", because there is no known next minor release date
            // This is only the case for the first minor of a planned future major version series.
```

Reply:
Updated